### PR TITLE
ISSUE-2272: Ingestion: Roles and Teams apis called only when ingesting users

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes :

Issue #2272 
Roles and Teams apis called only when ingesting users

#
### Type of change :

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] All new and existing tests passed.

#
### Reviewers

Ingestion: @harshach @ayush-shah @pmbrull

### CODE REVIEW

**Comment by pmbrull** [comment]
Hi, @ulixius9! Thanks for the PR 🚀  Running `make black` should fix the styling issue :)

File: ingestion/src/metadata/ingestion/sink/metadata_rest.py [hunk 1]
```diff
@@ -385,14 +384,6 @@ def write_ml_model(self, model: CreateMlModelEntityRequest):
             logger.error(err)
             self.status.failure(f"Model: {model.name}")
 
-    def _bootstrap_entities(self):
-        team_response = self.api_client.get("/teams")
-        for team in team_response["data"]:
-            self.team_entities[team["name"]] = team["id"]
```
Comment (mithmatt) [inline ingestion/src/metadata/ingestion/sink/metadata_rest.py:391]: Are these variables self.team_entities and self.role_entities used elsewhere in this file?

Comment (ulixius9) [reply ingestion/src/metadata/ingestion/sink/metadata_rest.py:391]: Yes, these are used in _create_team and _create_role methods respectively.

Comment (ulixius9) [reply ingestion/src/metadata/ingestion/sink/metadata_rest.py:391]: Also the write_users method which I've modified to include the logic of _bootstrap_entities uses these variables.

**Review by harshach — APPROVED** [review]

**Pull request merged**